server/src/main/java/com/genymobile/scrcpy/FakeContext.java

```diff
@@ -24,32 +24,26 @@ private FakeContext() {
 
     @Override
     public String getPackageName() {
-        if (Os.getuid() == 1000) {
-            return "android";
-        }
-        return PACKAGE_NAME;
+        return getPackageNameStatic();
     }
 
     @Override
     public String getOpPackageName() {
-        if (Os.getuid() == 1000) {
-            return "android";
-        }
-        return PACKAGE_NAME;
+        return getPackageNameStatic();
     }
 
     public static String getPackageNameStatic() {
         if (Os.getuid() == 1000) {
             return "android";
         }
-        return PACKAGE_NAME;
+        return PACKAGE_SHELL;
```

Comment:
We could probably assign the `PACKAGE_NAME` constant without a method:

```java
public static final String PACKAGE_NAME = Os.getuid() == 1000 ? "android" : "com.android.shell";
```

In practice, what is the problem if the package name is `"com.android.shell"` while the uid is 1000?

Reply:
At the very least the clipboard manager is entirely busted when UID and package don't match up - 

<details>
  <summary>Package com.android.shell does not belong to 1000</summary>
  
  ```
[server] ERROR: Could not invoke method
java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Method.invoke(Native Method)
        at com.genymobile.scrcpy.wrappers.ClipboardManager.getPrimaryClip(ClipboardManager.java:87)
        at com.genymobile.scrcpy.wrappers.ClipboardManager.getText(ClipboardManager.java:118)
        at com.genymobile.scrcpy.Device.getClipboardText(Device.java:285)
        at com.genymobile.scrcpy.Device.setClipboardText(Device.java:298)
        at com.genymobile.scrcpy.Controller.setClipboard(Controller.java:398)
        at com.genymobile.scrcpy.Controller.handleEvent(Controller.java:164)
        at com.genymobile.scrcpy.Controller.control(Controller.java:83)
        at com.genymobile.scrcpy.Controller.lambda$start$0$com-genymobile-scrcpy-Controller(Controller.java:91)
        at com.genymobile.scrcpy.Controller$$ExternalSyntheticLambda1.run(Unknown Source:4)
        at java.lang.Thread.run(Thread.java:1012)
Caused by: java.lang.SecurityException: Package com.android.shell does not belong to 1000
        at android.os.Parcel.createExceptionOrNull(Parcel.java:3011)
        at android.os.Parcel.createException(Parcel.java:2995)
        at android.os.Parcel.readException(Parcel.java:2978)
        at android.os.Parcel.readException(Parcel.java:2920)
        at android.content.IClipboard$Stub$Proxy.getPrimaryClip(IClipboard.java:387)
        ... 11 more
Caused by: android.os.RemoteException: Remote stack trace:
        at android.app.AppOpsManager.checkPackage(AppOpsManager.java:8803)
        at com.android.server.clipboard.ClipboardService.clipboardAccessAllowed(ClipboardService.java:1054)
        at com.android.server.clipboard.ClipboardService.clipboardAccessAllowed(ClipboardService.java:1040)
        at com.android.server.clipboard.ClipboardService.-$$Nest$mclipboardAccessAllowed(Unknown Source:0)
        at com.android.server.clipboard.ClipboardService$ClipboardImpl.getPrimaryClip(ClipboardService.java:461)

[server] ERROR: Could not invoke method
java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Method.invoke(Native Method)
        at com.genymobile.scrcpy.wrappers.ClipboardManager.setPrimaryClip(ClipboardManager.java:107)
        at com.genymobile.scrcpy.wrappers.ClipboardManager.setText(ClipboardManager.java:133)
        at com.genymobile.scrcpy.Device.setClipboardText(Device.java:308)
        at com.genymobile.scrcpy.Controller.setClipboard(Controller.java:398)
        at com.genymobile.scrcpy.Controller.handleEvent(Controller.java:164)
        at com.genymobile.scrcpy.Controller.control(Controller.java:83)
        at com.genymobile.scrcpy.Controller.lambda$start$0$com-genymobile-scrcpy-Controller(Controller.java:91)
        at com.genymobile.scrcpy.Controller$$ExternalSyntheticLambda1.run(Unknown Source:4)
        at java.lang.Thread.run(Thread.java:1012)
Caused by: java.lang.SecurityException: Package com.android.shell does not belong to 1000
        at android.os.Parcel.createExceptionOrNull(Parcel.java:3011)
        at android.os.Parcel.createException(Parcel.java:2995)
        at android.os.Parcel.readException(Parcel.java:2978)
        at android.os.Parcel.readException(Parcel.java:2920)
        at android.content.IClipboard$Stub$Proxy.setPrimaryClip(IClipboard.java:333)
        ... 10 more
  ```
</details>

Reply:
> We could probably assign the `PACKAGE_NAME` constant without a method:

Definitely yes, done